junifer/pipeline/update_meta_mixin.py

```diff
@@ -36,17 +36,35 @@ def update_meta(
         for k, v in vars(self).items():
             if not k.startswith("_"):
                 t_meta[k] = v
-        # Add "meta" to the step's local context dict
-        if "meta" not in input:
-            input["meta"] = {}
-        # Add step name
-        input["meta"][step_name] = t_meta
-        # Add step dependencies
-        if "dependencies" not in input["meta"]:
-            input["meta"]["dependencies"] = set()
-        # Update step dependencies
-        dependencies = getattr(self, "_DEPENDENCIES", set())
-        if dependencies is not None:
-            if not isinstance(dependencies, (set, list)):
-                dependencies = {dependencies}
-            input["meta"]["dependencies"].update(dependencies)
+        # Conditional for list dtype vals like Warp
+        if isinstance(input, list):
```

Comment:
same here, if it's not a list, make it a single element list and avoid duplicated code.